### PR TITLE
write query parameters to swagger definition

### PR DESCRIPTION
**Pull request by t-yuki**
It takes over from #199

### CODE REVIEW

File: examples/examplepb/a_bit_of_everything.swagger.json [hunk 1]
```diff
@@ -66,6 +66,197 @@
...
+            "name": "oneof_string",
+            "in": "query",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "name": "map_value",
+            "in": "query",
+            "required": false,
+            "type": "object"
```
Comment (t-yuki) [inline examples/examplepb/a_bit_of_everything.swagger.json]: map values is hard to process in query parameter. (some WAF handles it in `map_value[KEY]=VALUE`)
should we ignore it initially?

File: examples/examplepb/a_bit_of_everything.swagger.json [hunk 2]
```diff
@@ -57,8 +57,210 @@
...
+            "in": "query",
+            "required": false,
+            "type": "array",
+            "items": {
+              "type": "string",
+              "enum": [
+                "ZERO",
+                "ONE"
+              ],
+              "default": "ZERO"
```
Comment (achew22) [inline examples/examplepb/a_bit_of_everything.swagger.json]: Should the default be ["ZERO"]? or maybe []?

Comment (t-yuki) [inline examples/examplepb/a_bit_of_everything.swagger.json]: Agree. I unset default field.
Note that, swagger [petstore example](http://petstore.swagger.io/v2/swagger.json) has the array parameter with default field.
I wonder what it means but in protobuf, it should be nil.

`"parameters":[{"name":"status","in":"query","description":"Status values that need to be considered for filter","required":true,"type":"array","items":{"type":"string","enum":["available","pending","sold"],"default":"available"},"collectionFormat":"multi"}],`

**Comment by t-yuki** [comment]
ping @googlebot

**Comment by rilinor** [comment]
+1, We need this feature, merge it please!

**Comment by achew22** [comment]
@rgarcia, can you confirm a couple of things for me please before I merge this?

1.  You are the author of the commits that @t-yuki extended upon
1.  You're comfortable with the additional changes made
1.  To the best of your knowledge you've signed the CLA (https://github.com/grpc-ecosystem/grpc-gateway/pull/199 implies yes but I wanna double check)

**Comment by rgarcia** [comment]
@achew22 1. yes 2. yes 3. yes 😄

**Comment by achew22** [comment]
Huh, foiled by the CLI bot. Even on the command line I can't push past this. @yugui can we change the CLAbot to being advisory or non-blocking or whatever github calls it?

**Comment by t-yuki** [comment]
Should I re-push without rebase to avoid CLA bot fail?

**Comment by achew22** [comment]
We could be really tricky and merge the other PR then this one quickly as not to break anyone, hopefully. I just sent an email to yugui (not @'d on purpose cause I just sent her an email) about modifying the submission rules. Let's see how that plays out. Meet back here in 24 hours?

**Comment by yugui** [comment]
@achew22 I don't think we can make the bot advisory.  But I have granted admin rights of this project to you and @tmc.  So now you can ignore the status check at your own risk.

**Pull request merged**